frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/PushAction.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.actions;
+
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.dvcs.repo.VcsRepositoryManager;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ * </ul>
+ */
+public class PushAction extends AnAction implements DumbAware {
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent e) {
+    super.update(e);
+
+    Project project = e.getProject();
+    Option<GitRepository> selectedVcsRepository = ActionUtils.getSelectedVcsRepository(e);
+
+    boolean isSomeRepositoryPresent = project != null && !VcsRepositoryManager.getInstance(project).getRepositories().isEmpty();
+    boolean isSelectedVcsRepositoryPresent = selectedVcsRepository.isDefined();
+
+    e.getPresentation().setEnabledAndVisible(isSomeRepositoryPresent && isSelectedVcsRepositoryPresent);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
```

Comment:
Is it possible to make the `Force Push` option selected by default in the `VcsPushDialog` if needed (i.e. the current branch diverged from its remote counterpart)?

Reply:
This is not so simple since the action is located deeply and privately inside `VcsPushDialog` (`com.intellij.dvcs.push.ui.VcsPushDialog.ComplexPushAction#ComplexPushAction`). I'll try! :muscle:

Reply:
It's not obvious how to make `Force Push` a default option of our `PushDialog`. The main reason for that is the fact that `VcsPushDialog` uses private classes inside.
To achieve the goal we can:
- rewrite `VcsPushDialog`
- use reflection to modify `VcsPushDialog`

It is important to take consistency into account.

Reply:
issue created  #256